notebooks/classify/signature.py

```diff
@@ -0,0 +1,249 @@
+        """
+        for index in range(len(row)):
+            
+            currentFeature = columns[index]
+            value = row[index]
+            
+            # If the feature is missing we won't count it
+            if currentFeature not in presentFeatures:
+                continue
+                
+            if value not in featureDict[currentFeature]:
+                featureDict[currentFeature][value] = 1
+            else:
+                featureDict[currentFeature][value] += 1
+        return featureDict
+
+    def generate(self, binTable: pd.DataFrame, cleanTable: pd.DataFrame):
+        """[Function for generating data signatures from table given the presence/absence of data]
+
+        Args:
+            binTable (pd.DataFrame): [binarized dataframe generated from raw dataframe]
+            cleanTable (pd.DataFrame): [cleaned dataframe geneated from raw dataframe]
+
+        Returns:
+            [type]: [
+                signaturedict: dictionary
+                key: signature
+                value(s):
+                    count: # of features (int)
+                    presentFeatures: list of features (str)
+                    missingFeatures: list of features (str)
+                    featureDict:
+                        key: feature (str)
+                        values:
+                            dictionary of {(feature): {value}}  and their counts
+            ]
+        """
+
+        #print('Generating dictionary of signatures for table.')
+        columns = binTable.columns
+        signatureDict = {}
+        
+        for index, row in binTable.iterrows():
+            signature = ''.join(map(str, row.values.tolist()))
+            
+            # If this signature does not exist
+            if signature not in signatureDict:
+                
+                # Identify Present/Missing features
+                present, missing = self.getPresentColumns(signature, columns)
+                # Generate and update number of different data types in the feature dictionary
+                featureDict = {i: {} for i in present}
+                featureDict = self.countTypes(cleanTable.iloc[index], columns, present, featureDict)
+                
+                signatureDict[signature] = {
+                    'count': 1,
+                    'presentFeatures': present,
+                    'missingFeatures': missing,
+                    'featureDict': featureDict
+                }
+            else:
+                signatureDict[signature]['count'] += 1
+                signatureDict[signature]['featureDict'] = self.countTypes(cleanTable.iloc[index], columns, signatureDict[signature]['presentFeatures'], signatureDict[signature]['featureDict'])
+
+        return signatureDict
+
+    def generateSignatures(self, exactMatches: List[str] = [], regexes: List[str] = [], removeDuplicates:  bool = True, isInvariant: bool = True, removeNaN: bool = True):
```

Comment:
Might move this method to just after __init__ since it's really the main controlling method of the class.
name = generate_signatures

Using terms that a user will find familiar is also a good thing - e.g. "columns" in place of "features" - if you renamed the params something like "drop_columns", "drop_col_patterns" (regexes), drop_dups, drop_invar, drop_nan,
It would make the interface more accessible and consistent.